geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/io/LayerReaderFactory.scala

```diff
@@ -66,26 +68,26 @@ abstract class FilteringLayerReaderWrapper()
     keyType: String,
     layerName: String,
     zoom: Int
-  ): (JavaRDD[Array[Byte]], String) = {
+  ): TiledRasterRDD[_] = {
     val id = LayerId(layerName, zoom)
     val valueClass = getValueClass(id)
 
     (keyType, valueClass) match {
       case ("SpatialKey", "geotrellis.raster.Tile") => {
         val result = layerReader.read[SpatialKey, Tile, TileLayerMetadata[SpatialKey]](id)
-        PythonTranslator.toPython(tileToMultiband[SpatialKey](result))
+        new SpatialTiledRasterRDD(MultibandTileLayerRDD(tileToMultiband[SpatialKey](result), result.metadata))
```

Comment:
I was actually going to put that in a separate PR next. However, if you wanted it in this PR I can add it in.